model/demo_model.py

```diff
@@ -0,0 +1,208 @@
+import json
+from typing import List
+import View.AptLayout as AptView
+from View.time_slider import TimeSlider
+from model.abstract_model import Model
+from model.abstract_timer import ATimer
+from model.algorithm.abstract_mimicking_algorithm import MimickingAlgorithm
+from model.events.lamp_event import LampEvent
+from model.schedule import Schedule
+from datetime import datetime
+import time as t
+from model.events.lamp_action import LampAction
+
+class DemoModel(Model):
+    def __init__(self, scheduler : MimickingAlgorithm, timer: ATimer, start_time : datetime):
+        self.timer = timer
+        self.scheduler = scheduler
+        self.user_schedule = self.get_user_schedule()
+        self.algortihm_schedule = self.get_schedule()
+        
+        self.use_using_schedule = True
+        self.current_schedule = self.user_schedule
+        
+        self.start_time = start_time
+        self.current_hours = self.start_time.hour
+        self.current_minutes = self.start_time.minute
+        self.current_time = self.start_time
+        self.prev_time = self.start_time
+        self.observers = []
+
+        self.current_time_in_minutes = self.current_time.hour * 60 + self.current_time.minute
+            
+        self.current_active_lamps_list = self.create_active_lamp_list()
+        self.currently_active_lamps = self.current_active_lamps_list[self.current_time_in_minutes]
+        self.prev_current_active_lamps_list = self.current_active_lamps_list
+
+        # to turn on lamps that should be turned on, if given a start time that means that lights should be on
+        for lamp in self.currently_active_lamps:
+            self.publish_lamp_event(self.current_time,lamp,LampAction.ON)
+
+    def start(self):
+        self.timer.start()
+        self.timer.set_time(self.start_time)
```

Comment:
Do we always want to reset the time when stopping then starting?

Reply:
No maybe not, but does this make a difference to our demo? I can of course improve it if needed.

Reply:
Shouldn't make a difference to our demo, if we don't have the ability to pause in the Demo GUI